src/external/aave/StaticATokenV3.sol

```diff
@@ -0,0 +1,416 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity >=0.5.0;
+pragma experimental ABIEncoderV2;
+
+import {ILendingPool} from '../../interfaces/external/aave/ILendingPool.sol';
+import {IERC20} from "../../../lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IERC20Metadata} from "../../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IAToken} from '../../interfaces/external/aave/IAToken.sol';
+import {IRewardsController} from '../../interfaces/external/aave/IRewardsController.sol';
+import {ERC20} from "../../../lib/openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20} from "../../../lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import {WadRayMath} from './WadRayMath.sol';
+import {TokenUtils} from "../../libraries/TokenUtils.sol";
+
+/**
+ * @title StaticAToken updated to work with alchemix rewardCollector
+ * @dev Wrapper token that allows to deposit tokens on the Aave protocol and receive
+ * a token which balance doesn't increase automatically, but uses an ever-increasing exchange rate
+ * - Only supporting deposits and withdrawals
+ * @author Aave
+ **/
+contract StaticATokenV3 is ERC20 {
+  using SafeERC20 for IERC20;
+  using WadRayMath for uint256;
+
+  struct SignatureParams {
+    uint8 v;
+    bytes32 r;
+    bytes32 s;
+  }
+
+  bytes public constant EIP712_REVISION = bytes('1');
+  bytes32 internal constant EIP712_DOMAIN =
+    keccak256('EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)');
+  bytes32 public constant PERMIT_TYPEHASH =
+    keccak256('Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)');
+  bytes32 public constant METADEPOSIT_TYPEHASH =
+    keccak256(
+      'Deposit(address depositor,address recipient,uint256 value,uint16 referralCode,bool fromUnderlying,uint256 nonce,uint256 deadline)'
+    );
+  bytes32 public constant METAWITHDRAWAL_TYPEHASH =
+    keccak256(
+      'Withdraw(address owner,address recipient,uint256 staticAmount, uint256 dynamicAmount, bool toUnderlying, uint256 nonce,uint256 deadline)'
+    );
+
+  address public immutable REWARDCOLLECTOR;
```

Comment:
add a setter for this, as well as an admin that is the only one that can call the setter